docs/architecture-and-design.md

```diff
@@ -394,3 +400,62 @@ In order for a robot to move to the desired destination of a `MoveIntent`, the N
+
+Ultimately, we want a test suite that validates our [Plays](#plays) are generally doing the right thing. We might not care exactly where a robot receives the ball during a passing play, as long as the pass was successful overall. The solution to this problem is to use simulation to allow us to deterministically run our entire AI pipeline and validate behavior.
+
+The primary goals of this test system are:
+1. **Determinism**
+    * We need tests to pass or fail consistently
+2. **Test "ideal" behaviour**
+    * We want to test the logic in a "perfect world", where we don't care about all the exact limitations of our system in the real world with real physics. Eg. we don't care about modelling robot wheels slipping on the ground as we accelerate.
+
+## Simulated Integration Tests Architecture
+Fortunately, our abstractions and [Observer system](#observer-design-pattern) make it very easy to modify our system to support simulated integration tests. There are only 2 major changes we make to the system:
+1. Implement a [Backend](#backend) to handle the simulation
+2. Add another Observer to the system that will handle the "validation"
+
+The [SimulatorBackend](#simulator-backend) and [World State Validator](#world-state-validator) implement each of these changes respectively, and are further described in their own sections.
+
+### Simulator Backend
+The `SimulatorBackend` is simply another implementation of the [Backend](#backend) interface. It uses a physics library to simulate the various components of the [World](#world), like the [Ball](#ball) and [Robots](#robot). Like any [Backend](#backend), the `SimulatorBackend` publishes the state of the [World](#world) periodically.
+
+In order to achieve determinism, the `SimulatorBackend` will publish new [World](#world)'s with a fixed time increment (in [World](#world) time), and will wait to receive [Primitives](#primitives) before simulating and publishing the next [World](#world). This means that no matter how much faster or slower the simulation runs than the rest of the system, everyting will always happen "at the same speed" from the POV of the rest of the system, since each newly published [World](#world) will be a fixed amount of time newer than the last. See the section on [Component Connections and Determinism](#component-connections-and-determinism) for why this is important.
```

Comment:
nit:
```suggestion
In order to achieve determinism, the `SimulatorBackend` will publish new [World](#world)'s with a fixed time increment (in [World](#world) time), and will wait to receive [Primitives](#primitives) before simulating and publishing the next [World](#world). This means that no matter how much faster or slower the simulation runs than the rest of the system, everything will always happen "at the same speed" from the POV of the rest of the system, since each newly published [World](#world) will be a fixed amount of time newer than the last. See the section on [Component Connections and Determinism](#component-connections-and-determinism) for why this is important.
```